autonomic-administration-plugin/src/test/java/br/com/autonomiccs/autonomic/administration/plugin/AdministrationAgentTest.java

```diff
@@ -0,0 +1,665 @@
+        verifyReceiveClusterToBeManaged(1, 1);
+    }
+
+    @Test
+    public void searchClusterToBeManagedTestReturnedClusterNull() {
+        ClusterVO cluster = Mockito.mock(ClusterVO.class);
+        Mockito.doReturn(null).when(administrationAgent).getClusterToManage(Mockito.any(ClusterAdministrationHeuristicAlgorithm.class));
+
+        Long result = administrationAgent.searchClusterToBeManaged();
+
+        Assert.assertNull(result);
+        verifySearchClusterToBeManagedCalledMethods(cluster, 0);
+    }
+
+    @Test
+    public void searchClusterToBeManagedTestReturnedClusterNotNull() {
+        ClusterVO cluster = Mockito.mock(ClusterVO.class);
+        Mockito.doReturn(cluster).when(administrationAgent).getClusterToManage(Mockito.any(ClusterAdministrationHeuristicAlgorithm.class));
+
+        Long result = administrationAgent.searchClusterToBeManaged();
+
+        Assert.assertNotNull(result);
+        verifySearchClusterToBeManagedCalledMethods(cluster, 1);
+    }
+
+    @Test
+    public void getClusterToManageTestEmptyEnabledZonesList() {
+        Mockito.doReturn(new ArrayList<>()).when(zoneService).listAllZonesEnabled();
+
+        ClusterVO result = administrationAgent.getClusterToManage(clusterAdministrationHeuristicAlgorithm);
+        Assert.assertNull(result);
+        verifyGetClusterToManageCalledMethods(0);
+    }
+
+    @Test
+    public void getClusterToManageTestNotEmptyEnabledZonesList() {
+        List<DataCenterVO> zones = new ArrayList<>();
+        zones.add(new DataCenterVO(0, "name", "description", "", "", "", "", "", "", 0l, NetworkType.Basic, "", ""));
+        ClusterVO mockedCluster = Mockito.mock(ClusterVO.class);
+
+        Mockito.doReturn(zones).when(zoneService).listAllZonesEnabled();
+        Mockito.doReturn(mockedCluster).when(administrationAgent).getClusterToManageFromZone(Mockito.any(ClusterAdministrationHeuristicAlgorithm.class), Mockito.anyLong());
+
+        ClusterVO result = administrationAgent.getClusterToManage(clusterAdministrationHeuristicAlgorithm);
+
+        Assert.assertEquals(mockedCluster, result);
+        Assert.assertNotNull(result);
+        verifyGetClusterToManageCalledMethods(1);
+    }
+
+    @Test
+    public void getClusterToManageFromZoneTestListOfClustersEmpty() {
+        List<ClusterVO> clusters = new ArrayList<>();
+        Mockito.doReturn(clusters).when(clusterService).listAllClustersOnZone(Mockito.anyLong());
+
+        administrationAgent.getClusterToManageFromZone(clusterAdministrationHeuristicAlgorithm, null);
+
+        verifyGetClusterToManageFromZoneCalledMethods(0, 0, 0);
+    }
+
+    @Test
+    public void getClusterToManageFromZoneTestListOfClustersNotEmptyAndClusterIsNotAdministrated() {
+        List<ClusterVO> clusters = createClusters(null);
+        getClusterToManageFromZoneReturnedValuesSetup(clusters, false, true);
+
+        administrationAgent.getClusterToManageFromZone(clusterAdministrationHeuristicAlgorithm, null);
```

Comment:
If I read the method's name right, now it should return your configured cluster, right?